### PR TITLE
Add Liger Kernel Support for qwen3_vl

### PR DESCRIPTION
**Pull request by thad0ctor**
# Add Liger Kernel Support for qwen3_vl

# Description

This PR adds native support for Qwen3 Vision-Language (non-MOE) models with Liger kernel optimizations. The implementation aligns with the upstream Liger Kernel commit [6cd9fd64](https://github.com/linkedin/Liger-Kernel/commit/6cd9fd64a51e26b27a7dc9cc33c9b7ed1ac567cc) that introduced `apply_liger_kernel_to_qwen3_vl`.

## Key Changes:

1. **Added Qwen3-VL Support in Liger Plugin** (`src/axolotl/integrations/liger/plugin.py`):
   - Import `apply_liger_kernel_to_qwen3_vl` from `liger_kernel.transformers`
   - Add configuration case for `qwen3_vl` model type
   - Include all supported Liger optimizations with proper parameter mapping

  2. **Parameter Mapping**:
     - `rope`: Rotary position embeddings optimization
     - `cross_entropy`: Standard cross-entropy loss optimization (mutually
  exclusive with fused_linear_cross_entropy)
     - `fused_linear_cross_entropy`: Memory-efficient fused implementation
  - combines final linear layer with cross-entropy loss
     - `rms_norm`: RMSNorm layer optimization
     - `swiglu`: SwiGLU activation optimization (note: qwen3_vl uses
  `swiglu` parameter name instead of `glu_activation`)

## Motivation and Context

**Problem:** Qwen3_vl models were not supported by Axolotl's Liger integration, resulting in the warning:
```
[WARNING] [axolotl.integrations.liger.plugin] Unsupported model config type: qwen3_vl. Liger not applied.
```

This prevented users from leveraging Liger's performance optimizations when training Qwen3-VL models.

**Solution:** The fix adds explicit support for the `qwen3_vl` model configuration type, enabling Liger optimizations that are compatible with vision-language models from the Liger upstream.

## How has this been tested?

**Testing Approach:**
1. Verified that the warning no longer appears when using `model_config_type: qwen3_vl`, trainging and eval steps are successful
2. Confirmed that Liger optimizations are correctly applied during model initialization
3. Checked compatibility with existing Liger configuration options
4. Ensured parameter mapping matches upstream Liger implementation
5. Witnessed presence of liger kernels in PyTorch compiler (Dynamo)

**Configuration Example:**
```yaml
base_model: Qwen/Qwen3-VL-4B-Instruct
model_config_type: qwen3_vl

# Liger optimizations
liger_rope: true
liger_rms_norm: true
liger_fused_linear_cross_entropy: true
liger_glu_activation: true  # Maps to swiglu parameter
liger_use_token_scaling: true  # Forces token scaling in FLC
```

## Screenshots (if appropriate)
N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Summary by CodeRabbit

* **New Features**
  * Added Qwen3 Vision-Language model support with LIGER kernel optimizations.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by thad0ctor** [comment]
@coderabbitai generate docstrings

**Comment by NanoCode012** [comment]
Hey, thanks for the PR. one question, have you tried updating to `liger-kernel==0.6.4`? The current locked version 0.6.3 doesn't have qwen3_vl, hence your error above. I think if you use the new version, it should just work.

**Comment by thad0ctor** [comment]
> Hey, thanks for the PR. one question, have you tried updating to `liger-kernel==0.6.4`? The current locked version 0.6.3 doesn't have qwen3_vl, hence your error above. I think if you use the new version, it should just work.

I was running 0.6.4  when I tested it and got the error, something in the flow (torch I think) was pulling qwen3_vl from the model config and causing it to not be initialized

**Comment by NanoCode012** [comment]
> > Hey, thanks for the PR. one question, have you tried updating to `liger-kernel==0.6.4`? The current locked version 0.6.3 doesn't have qwen3_vl, hence your error above. I think if you use the new version, it should just work.
> 
> I was running 0.6.4 when I tested it and got the error, something in the flow (torch I think) was pulling qwen3_vl from the model config and causing it to not be initialized

that's weird. I checked their 0.6.4 code and it should by default call qwen3_vl apply function already. These extra ones were added because the upstream repo did not support these model types and we had to add custom patches.

**Comment by thad0ctor** [comment]
you may be right looking at the trace, I could have had my venvs messed up in testing and used an older version.  I'm mid training now but will test and follow up.

**Comment by NanoCode012** [comment]
> you may be right looking at the trace, I could have had my venvs messed up in testing and used an older version. I'm mid training now but will test and follow up.

How did it go?

**Comment by thad0ctor** [comment]
> > you may be right looking at the trace, I could have had my venvs messed up in testing and used an older version. I'm mid training now but will test and follow up.
> 
> How did it go?

You were right, full support natively when checking with the right venv.  Closing PR now

**Comment by thad0ctor** [comment]
Closing PR
